pkg/upgraders/healthcheckstep.go

```diff
@@ -40,8 +40,12 @@ func (c *clusterUpgrader) PreUpgradeHealthCheck(ctx context.Context, logger logr
 	if c.upgradeConfig.Spec.CapacityReservation {
 		ok, err := c.scaler.CanScale(c.client, logger)
 		if !ok || err != nil {
+			// HealthCheck should not fail in this senario. Have to inform the customer about the upgrade delay.
 			c.metrics.UpdateMetricHealthcheckFailed(c.upgradeConfig.Name, metrics.DefaultWorkerMachinepoolNotFound)
-			healthCheckFailed = append(healthCheckFailed, "CapacityReservationHealthcheckFailed")
+			err = c.notifier.Notify(notifier.MuoStateDelayed)
+			if err != nil {
+				return false, err
```

Comment:
If it fails to Notify, will it block the next pre-health check?

Reply:
NotificationEventSent metric set in the end just before nil error return, hence if there is an error it will return before the metrics gets update.
https://github.com/openshift/managed-upgrade-operator/blob/master/pkg/eventmanager/eventmanager.go#L159